padertorch/data/segment.py

```diff
@@ -0,0 +1,645 @@
+            to_segment_length, length, shift, anchor=self.anchor,
+            mode='constant', rng=rng
+        )
+
+        segmented = {key: segment(
+            signal, length=length, shift=shift, rng=rng, axis=axis[i],
+            anchor=anchor, padding=self.padding, mode='constant'
+        ) for i, (key, signal) in enumerate(to_segment.items())}
+        return boundaries, segmented
+
+    def get_to_segment_keys(self, example):
+        if self.include is None:
+            return [
+                key for key, value in example.items()
+                if key not in self.exclude and
+                isinstance(value, (np.ndarray, torch.Tensor))
+            ]
+        else:
+            to_segment_keys = [
+                key for key in example.keys() if
+                key not in self.exclude and
+                any([key.startswith(include_key)
+                     for include_key in self.include])
+            ]
+            assert all([
+                any([key.startswith(include_key) for key in to_segment_keys])
+                for include_key in self.include
+            ]), ('For some keys in include_keys no associated key was found '
+                 'in the example', self.include, to_segment_keys)
+            return to_segment_keys
+
+    def get_axis_list(self, to_segment_keys):
+        if isinstance(self.axis, int):
+            return [self.axis] * len(to_segment_keys)
+        elif isinstance(self.axis, dict):
+            axis = self.axis.copy()
+            axis.update({
+                to_segment_key: axis for to_segment_key in to_segment_keys
+                for org_key, axis in self.axis.items()
+                if to_segment_key.startswith(org_key)
+            })
+            assert all([key in axis.keys() for key in to_segment_keys]), (
+                f'The dictionary for axis did not include keys for all'
+                f'segment arrays. axis keys: {self.axis.keys()},'
+                f'segment array keys {to_segment_keys}'
+            )
+            return [axis[key] for key in to_segment_keys]
+        elif isinstance(self.axis, list):
+            assert len(self.axis) == len(to_segment_keys), (
+                f'The list for axis does not include a axis for each'
+                f'segment array. axis: {self.axis}, '
+                f'segment array keys {to_segment_keys}'
+            )
+            return self.axis
+        else:
+            raise TypeError('This should never be reached', self.axis)
+
+
+def _get_rand_int(rng, *args, **kwargs):
+    if hasattr(rng, 'randint'):
+        return rng.randint(*args, **kwargs)
+    elif hasattr(rng, 'integers'):
+        return rng.integers(*args, **kwargs)
+    elif isinstance(rng, callable):
+        return rng(*args, **kwargs)
+    else:
+        raise TypeError('Unknown random generator used', rng)
+
+
+def get_anchor(
+        num_samples: int, length: int, shift: int = None,
+        mode: str = 'left', rng=np.random
+):
```

Comment:
I just noticed that the return type is not annotated. Can you add that?

Reply:
done